### PR TITLE
Add overriden_class to other backends

### PR DESCRIPTION
**Pull request by joaosferreira**
Picking up where #61 left off by extending class overriding to other backends, more specifically, Dask and Sparse.

### CODE REVIEW

File: unumpy/sparse_backend.py [hunk 1]
```diff
@@ -23,14 +23,34 @@ def array(x, *args, **kwargs):
     return sparse.COO.from_numpy(np.asarray(x))
 
 
+def overridden_class(self):
+    if self is ndarray:
+        return sparse._coo.core.COO
```
Comment (hameerabbasi) [inline unumpy/sparse_backend.py]:
```suggestion
        return sparse.SparseArray
```

File: .gitignore [hunk 2]
```diff
@@ -25,3 +25,4 @@ build/
 .eggs/
 dask-worker-space/
 *.ipynb
+default.profraw
```
Comment (hameerabbasi) [inline .gitignore:28]: Thanks for this!

File: unumpy/sparse_backend.py (hunk 1 above)
Comment (joaosferreira) [reply unumpy/sparse_backend.py]: I forced pushed the suggested change.

**Review by hameerabbasi — APPROVED** [review]
LGTM

**Pull request merged**